crates/services/block_aggregator_api/src/blocks/importer_and_db_source/sync_service.rs

```diff
@@ -0,0 +1,171 @@
+use crate::blocks::{
+    BlockSourceEvent,
+    importer_and_db_source::BlockSerializer,
+};
+use fuel_core_services::{
+    RunnableService,
+    RunnableTask,
+    StateWatcher,
+    TaskNextAction,
+    try_or_continue,
+    try_or_stop,
+};
+use fuel_core_storage::{
+    self,
+    StorageInspect,
+    tables::{
+        FuelBlocks,
+        Transactions,
+    },
+};
+use fuel_core_types::{
+    fuel_tx::{
+        Transaction,
+        TxId,
+    },
+    fuel_types::BlockHeight,
+};
+use tokio::sync::mpsc::Sender;
+
+pub struct SyncTask<Serializer, DB> {
+    serializer: Serializer,
+    block_return_sender: Sender<BlockSourceEvent>,
+    db: DB,
+    next_height: BlockHeight,
+    maybe_stop_height: Option<BlockHeight>,
+    new_ending_height: tokio::sync::oneshot::Receiver<BlockHeight>,
```

Comment:
Why is it a `oneshot` channel? You only can read it once. And it is not clear to me how do you want to use it=)

Reply:
I agree. It's complicated. But I want to avoid some race condition of telling it the height, but then missing one height from the importer.